engine/src/test/java/com/datatorrent/stram/client/StramClientUtilsTest.java

```diff
@@ -162,75 +169,94 @@ public InetSocketAddress getSocketAddr(String name, String defaultAddress, int d
   {
-    Configuration conf = new YarnConfiguration(new Configuration(false));
+    Configuration conf = new Configuration(false);
     conf.set(StramClientUtils.DT_DFS_ROOT_DIR, "/a/b/c");
     conf.setBoolean(StramUserLogin.DT_APP_PATH_IMPERSONATED, false);
 
+    UserGroupInformation testUser = UserGroupInformation.createUserForTesting("testUser1", new String[]{""});
+    spy(UserGroupInformation.class);
+    doReturn(testUser).when(UserGroupInformation.class, "getLoginUser");
+
     FileSystem fs = FileSystem.newInstance(conf);
     Path path = StramClientUtils.getApexDFSRootDir(fs, conf);
     Assert.assertEquals("file:/a/b/c", path.toString());
+    // verify conf properties for DT_DFS_ROOT_DIR and APEX_APP_DFS_ROOT_DIR
+    Assert.assertEquals("/a/b/c", conf.get(StramClientUtils.DT_DFS_ROOT_DIR));
+    Assert.assertNull(conf.get(StramClientUtils.APEX_APP_DFS_ROOT_DIR));
   }
 
   /**
    * apex.dfsRootDirectory set: absolute path e.g. /x/y/z
-   * @throws IOException
+   * @throws Exception
    *
    */
   @Test
-  public void getApexDFSRootDirAbsPath() throws IOException
+  public void getApexDFSRootDirAbsPath() throws Exception
   {
-    Configuration conf = new YarnConfiguration(new Configuration(false));
+    Configuration conf = new Configuration(false);
     conf.set(StramClientUtils.APEX_APP_DFS_ROOT_DIR, "/x/y/z");
     conf.setBoolean(StramUserLogin.DT_APP_PATH_IMPERSONATED, false);
 
-    FileSystem fs = FileSystem.newInstance(conf);
     UserGroupInformation testUser = UserGroupInformation.createUserForTesting("testUser1", new String[]{""});
-    UserGroupInformation.setLoginUser(testUser);
+    spy(UserGroupInformation.class);
+    doReturn(testUser).when(UserGroupInformation.class, "getLoginUser");
+
+    FileSystem fs = FileSystem.newInstance(conf);
     Path path = StramClientUtils.getApexDFSRootDir(fs, conf);
     Assert.assertEquals(fs.getHomeDirectory() + "/datatorrent", path.toString());
+    // verify conf properties for DT_DFS_ROOT_DIR and APEX_APP_DFS_ROOT_DIR
+    Assert.assertNull(conf.get(StramClientUtils.DT_DFS_ROOT_DIR));
+    Assert.assertEquals("/x/y/z", conf.get(StramClientUtils.APEX_APP_DFS_ROOT_DIR));
   }
 
   /**
    * apex.dfsRootDirectory set: absolute path with scheme e.g. file:/p/q/r
-   * @throws IOException
+   * @throws Exception
    *
    */
   @Test
-  public void getApexDFSRootDirScheme() throws IOException
+  public void getApexDFSRootDirScheme() throws Exception
   {
-    Configuration conf = new YarnConfiguration(new Configuration(false));
+    Configuration conf = new Configuration(false);
     conf.set(StramClientUtils.APEX_APP_DFS_ROOT_DIR, "file:/p/q/r");
     conf.setBoolean(StramUserLogin.DT_APP_PATH_IMPERSONATED, false);
 
-    FileSystem fs = FileSystem.newInstance(conf);
     UserGroupInformation testUser = UserGroupInformation.createUserForTesting("testUser1", new String[]{""});
-    UserGroupInformation.setLoginUser(testUser);
+    spy(UserGroupInformation.class);
```

Comment:
Can it be moved to `@Before` or `@BeforeClass`?

Reply:
No. Not all tests need that. But I will refactor this into a function that will be called from all the affected functions.

Reply:
@vrozov Let me know your comments on the latest changes and the current state of this PR. The PR shows "vrozov requested changes" and you had a -1 based on CI issues (not related to this PR).

Reply:
Even though not all test use UserGroupInformation, will it affect them? Note that you mock UserGroupInformation for the entire unit test, not only for the tests that need it.

Reply:
Yes, I am using PrepareForTest and RunWith annotations at class level because they could not be made to work at method level. There is more than one way, but I didn't think it was worth the time for me to refactor to address the `@Before` suggestion esp. at this stage.

Reply:
The current way gives wrong impression that UserGroupInformation won't be modified unless `setupLoginTestUser()` is invoked. Also, should not it be static and `@BeforeClass`?